src/main/controllers/AgreeingDocumentsForHearingController.ts

```diff
@@ -0,0 +1,122 @@
+import { FormContent, FormFields } from '../definitions/form';
+import { AnyRecord } from '../definitions/util-types';
+
+import { agreeingDocumentsForHearingErrors } from './helpers/ErrorHelpers';
+import { assignFormData, getPageContent } from './helpers/FormHelpers';
+import { getLanguageParam } from './helpers/RouterHelpers';
+
+export default class AgreeingDocumentsForHearingController {
+  private readonly form: Form;
+
+  private readonly agreeingDocumentsForHearingContent: FormContent = {
+    fields: {
+      bundlesRespondentAgreedDocWith: {
+        classes: 'govuk-radios',
+        id: 'bundlesRespondentAgreedDocWith',
+        type: 'radios',
+        label: (l: AnyRecord): string => l.legend,
+        labelSize: 'l',
+        labelHidden: false,
+        values: [
+          {
+            name: 'bundlesRespondentAgreedDocWith',
+            label: (l: AnyRecord): string => l.yes,
+            value: AgreedDocuments.YES,
+          },
+          {
+            name: 'bundlesRespondentAgreedDocWith',
+            label: (l: AnyRecord): string => l.agreed,
+            value: AgreedDocuments.AGREEDBUT,
+            subFields: {
+              bundlesRespondentAgreedDocWithBut: {
+                id: 'bundlesRespondentAgreedDocWithBut',
+                name: 'bundlesRespondentAgreedDocWithBut',
+                type: 'textarea',
+                hint: (l: AnyRecord): string => l.agreedTextLabel,
+                classes: 'govuk-textarea',
+                attributes: { maxLength: 2500 },
+                validator: isFieldFilledIn,
+              },
+            },
+          },
+          {
+            name: 'bundlesRespondentAgreedDocWith',
+            label: (l: AnyRecord): string => l.notAgreed,
+            value: AgreedDocuments.NOTAGREED,
+            subFields: {
+              bundlesRespondentAgreedDocWithNo: {
+                id: 'bundlesRespondentAgreedDocWithNo',
+                name: 'bundlesRespondentAgreedDocWithNo',
+                type: 'textarea',
+                hint: (l: AnyRecord): string => l.notAgreedTextLabel,
+                classes: 'govuk-textarea',
+                attributes: { maxLength: 2500 },
+                validator: isFieldFilledIn,
+              },
+            },
+          },
+        ],
+        validator: isFieldFilledIn,
+      },
+    },
+    submit: {
+      text: l => l.continue,
+    },
+  };
+
+  constructor() {
+    this.form = new Form(<FormFields>this.agreeingDocumentsForHearingContent.fields);
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    const { userCase } = req.session;
+
+    const selectedRadio = req.body.bundlesRespondentAgreedDocWith;
+    userCase.bundlesRespondentAgreedDocWith = selectedRadio;
+    let agreedButText = req.body.bundlesRespondentAgreedDocWithBut;
+    let notAgreedText = req.body.bundlesRespondentAgreedDocWithNo;
+
+    if (selectedRadio === AgreedDocuments.AGREEDBUT && agreedButText !== undefined) {
```

Comment:
Probably don't need to check for undefined text as there's already a validator on it. Same for other if

Reply:
Removed